src/idc/idc.c

```diff
@@ -93,6 +94,10 @@ int idc_wait_in_blocking_mode(uint32_t target_core, bool (*cond)(int))
 		IDC_TIMEOUT / 1000;
 
 	while (!cond(target_core)) {
+
+		/* spin here so other core can access IO and timers freely */
+		idelay(8192);
```

Comment:
After quick debug check this looks more complex, actually it is booting fine (trace point 4000 for secondary core too) and entered task_main_secondary_core() successfully

Reply:
yeah, was thing that too - we now have aloop here that lock and unlocks (with IRQs OFF) a lot. We need to clean this up so we
1) keep the relax
2) split the platform_timer_get() into locked and unlocked versions (scheduler uses locked, trace uses unlocked). This will use unlocked.

Reply:
Fwiw, the mailbox and timer IP reads are on a shared bus (that all cores must use), so this should actually speed up other cores booting since the IO bus has less traffic to block other core IO.

Reply:
@abonislawski difficult to see this one in the logs ? We could have a race here as I've seen red/green in the reulst on different runs. Added more trace to show any errors.

Reply:
@slawblauciak @abonislawski there is nothing in the logs showing why playback fails on the multicore tests. I can only assume we have something fishy going on with the locks - disabling for CI validation.

Reply:
All multicore/multkernel test pass when the locking is removed ! Lets try per core locking as we should only ever have one user of the timer.

Reply:
@mwasko @slawblauciak @abonislawski @keyonjie fyi - this now passes CI now. Multicore would race and mainly show red on CI when IRQs were globally OFF for timer get/set. I will cleanup and merge it tomorrow.

Reply:
guys I've kept recover cycles high here at original value. We can optimise as more debug info is collected.